### ISSUE TITLE
[feature] Add support for cable splicing

### ISSUE THREAD

**Issue by formatc1702** [formatc1702/WireViz#26]
Possibly using a "pseudo-connector" linking e.g. one wire from the left and two wires from the right.

Proof-of-concept:

```
connectors:
  X1:
    pincount: 5
  X2:
    pincount: 8
  S1:
    category: ferrule # TODO: splice
    pincount: 1
    type: Solderjoint + Tape
    notes: splice
  S2:
    category: ferrule # TODO: splice
    pincount: 1
    type: Solderjoint + Tape
    notes: splice

cables:
  W1: &template_wire
    category: bundle
    wirecount: 3
    gauge: 0.25
    length: 1
    show_name: false
    colors: [PK]
  W2:
    <<: *template_wire
    wirecount: 2
    length: 0.8
    colors: [TQ, YE]
  W3:
    <<: *template_wire
    wirecount: 5
    length: 0.2
    colors: [TQ, TQ, YE, YE, YE]

connections:
  -
    - X1: [1-3]
    - W1: [1-3]
    - X2: [1-3]
  -
    - X1: [4,5]
    - W2: [1,2]
  -
    - W2: 1
    - S1: 1
  -
    - W2: 2
    - S2: 1
  -
    - S1: [1,1]
    - W3: [1,2]
    - X2: [4,5]
  -
    - S2: [1,1,1]
    - W3: [3,4,5]
    - X2: [6,7,8]
```

![test](https://user-images.githubusercontent.com/2447251/85919501-d8691c00-b86b-11ea-8484-63335d198691.png)

**Comment by kvid** [comment]
Really nice! I guess a final solution can avoid the duplicated text "Solderjoint + Tape" inside the splice frame. :-)

**Comment by formatc1702** [comment]
Yes, this is just an artifact of the way ferrules are implemented now, since they initially were designed to have only one pin, and one wire going into it... splicing is on the roadmap, though!

I'm not quite sure why GraphViz makes the topmost wire bend like that, though... that's something I have little control over...

**Comment by formatc1702** [comment]
I refactores the entire ferrule/splice section, the bug with the double label is fixed, and splices now look good regardless of whether few or many attributes are specified.

![test](https://user-images.githubusercontent.com/2447251/86538600-3d211800-bef7-11ea-8cf5-96c69dc97aea.png)

I am considering to change the `ferrule` category name to `simple`, to encompass both ferrules, splices, and any other simple connectors that don't have a designator and essentially only have one port (electrically speaking, at least).

**Comment by kvid** [comment]
I suggest this also can be used for any product to connect several wires in one physical point, e.g. a Wire Nut, as mentioned in issue https://github.com/formatc1702/WireViz/issues/45#issuecomment-654097778

**Comment by formatc1702** [comment]
I consider this completed and have included some of the thoughts in #67.
